frontend/web/components/release-pipelines/CreateReleasePipeline.tsx

```diff
@@ -275,7 +280,7 @@ function CreateReleasePipeline() {
                       onAddStage={() =>
                         setPipelineData((prev) => ({
                           ...prev,
-                          stages: prev.stages.concat([blankStage]),
+                          stages: prev.stages.concat([NEW_PIPELINE_STAGE]),
```

Comment:
```suggestion
                          stages: prev.stages.push(NEW_PIPELINE_STAGE),
```

Reply:
Concat is immutable while pushing is mutable, I'd prefer using immutability in those cases